ArduCopter/events.cpp

```diff
@@ -105,18 +105,38 @@ void Copter::failsafe_gcs_check()
     // update state, log to dataflash
     set_failsafe_gcs(true);
     Log_Write_Error(ERROR_SUBSYSTEM_FAILSAFE_GCS, ERROR_CODE_FAILSAFE_OCCURRED);
+    gcs_send_text(MAV_SEVERITY_WARNING,"Lost contact with GCS!");
 
     // clear overrides so that RC control can be regained with radio.
     hal.rcin->clear_overrides();
     failsafe.rc_override_active = false;
 
     if (should_disarm_on_failsafe()) {
         init_disarm_motors();
-    } else {
-        if (control_mode == AUTO && g.failsafe_gcs == FS_GCS_ENABLED_CONTINUE_MISSION) {
-            // continue mission
-        } else if (g.failsafe_gcs != FS_GCS_DISABLED) {
-            set_mode_RTL_or_land_with_pause(MODE_REASON_GCS_FAILSAFE);
+    } else {        
+        switch (g.failsafe_gcs) {
+            case FS_GCS_ENABLED_CONTINUE_MISSION:
+                if (control_mode != AUTO) {
+                    set_mode_RTL_or_land_with_pause(MODE_REASON_GCS_FAILSAFE);
+                }
+                break;
+            case FS_GCS_ENABLED_GUIDED:
+                if (control_mode != GUIDED) {
+                    break;
+                }
```

Comment:
these two fall throughs need to be explicitly commented to prevent misinterpretation and someone "fixing" it later

// FALL THROUGH